src/Nethermind/Nethermind.Evm/Precompiles/Stateful/BeaconStateRootPrecompile.cs

```diff
@@ -0,0 +1,66 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Int256;
+using Nethermind.State;
+
+namespace Nethermind.Evm.Precompiles.Stateful;
+public class BeaconStateRootPrecompile : IPrecompile
+{
+    private readonly IWorldState _state;
+    public static Address Address { get; } = Address.FromNumber(0x0B);
+    public static UInt256 HISTORICAL_ROOTS_LENGTH = 98304;
+    private BeaconStateRootPrecompile(IWorldState state)
+    {
+        _state = state;
+    }
+
+    public static IPrecompile Instance(IWorldState state) => new BeaconStateRootPrecompile(state);
+
+    private byte[] SloadFromStorage(UInt256 index)
```

Comment:
Pass `UInt256 ` via in 
```suggestion
    private byte[] SloadFromStorage(in UInt256 index)
```